src/utils/contractHelpers.ts

```diff
@@ -50,10 +50,14 @@ import easterNftAbi from 'config/abi/easterNft.json'
 import cakeVaultAbi from 'config/abi/cakeVault.json'
 import predictionsAbi from 'config/abi/predictions.json'
 import chainlinkOracleAbi from 'config/abi/chainlinkOracle.json'
+import { DEFAULT_GAS_PRICE } from 'config'
+import { getSettings } from './settings'
 
-const getContract = (abi: any, address: string, web3?: Web3) => {
+const getContract = (abi: any, address: string, web3?: Web3, account?: string) => {
```

Comment:
I opted for passing in the account because getting the user settings from the web3 object is

1. Not guaranteed to be there
2. Would require (as far as I know) for the function to be async. We would then have to update the entire code base